### ISSUE TITLE
Fix roslyn warnings in Analyzer packages

### ISSUE THREAD

**Issue by JunTaoLuo** [dotnet/aspnetcore#22090]
There are a few warnings in our analyzer packages that need to be fixed:

```
ComponentFacts.cs(27,24): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentFacts.cs(27,74): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentFacts.cs(43,54): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentFacts.cs(58,74): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
InternalUsageAnalyzer.cs(129,35): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentParameterAnalyzer.cs(30,57): warning RS1026: Enable concurrent execution [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentParameterAnalyzer.cs(30,57): warning RS1025: Configure generated code analysis [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentParameterUsageAnalyzer.cs(26,57): warning RS1026: Enable concurrent execution [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
InternalUsageAnalyzer.cs(158,35): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentFacts.cs(87,54): warning RS1024: Compare symbols correctly [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
ComponentInternalUsageDiagnosticAnalzyer.cs(31,57): warning RS1026: Enable concurrent execution [C:\gh\aspnetcore\src\Components\Analyzers\src\Microsoft.AspNetCore.Components.Analyzers.csproj]
```

Currently, these warnings are disabled in https://github.com/dotnet/aspnetcore/blob/b9899d9649d43a3816ecc0f30669551e0492dbcb/Directory.Build.props#L74-L75. We should fix the warnings and stop ignoring these in Directory.Build.props.

FYI @NTaylorMullen to assign since I'm not too sure who would own these.

**Comment by NTaylorMullen** [comment]
@pranavkm you still own analyzers right?

**Comment by mkArtakMSFT** [comment]
@dougbu, @JunTaoLuo is this already fixed or still an issue?

**Comment by JunTaoLuo** [comment]
Still an issue. See https://github.com/dotnet/aspnetcore/blob/cd9da1c3503d89e7f42b7ee780416eb1d4036b84/Directory.Build.props#L109

**Comment by dougbu** [comment]
Suppressing these warnings **might** be something we could live with if that's required. We have lots of `<NoWarn>` settings sprinkled through our `msbuild` files. But, most of those are about doc comments i.e. `CS1591`. The question here is about resources needed and priority of removing the `<NoWarn>` setting to address the underlying issue(s).

**Comment by JunTaoLuo** [comment]
I believe when I chatted with @NTaylorMullen he thinks that these warnings are actual bugs the need to be fixed. I used NoWarn to unblock the repo consolidation so the underlying issues still need to be resolved. But I'll defer to him to confirm that's the case and if so, the priority of this work item.

**Comment by NTaylorMullen** [comment]
The warnings help protect us against edge-case bugs. Priority, low priority because it's been this way forever

**Comment by ajaybhargavb** [comment]
[Here is a commit](https://github.com/dotnet/aspnetcore/pull/23630/commits/5e67d69feb8d3072754e31ecadb0e3d164765869) that fixes some of these warnings as an example.

**Comment by JunTaoLuo** [comment]
Looks like all the warnings have been addressed closing.

**Comment by ajaybhargavb** [comment]
Unfortunately not. The commit I referenced was only a subset. I had to add back the `NoWarn`s because I didn't want to handle all of the warnings there.
